pkgs/by-name/co/cosmic-comp/package.nix

```
@@ -56,11 +71,13 @@ rustPlatform.buildRustPackage {
   # These libraries are only used by the X11 backend, which will not
   # be the common case, so just make them available, don't link them.
   postInstall = ''
-    wrapProgram $out/bin/cosmic-comp \
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
+    wrapProgramArgs=(--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
         xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr
-      ]} \
-      --prefix PATH : ${lib.makeBinPath [ xwayland ]}
+    ]})
+  '' + lib.optionalString useXWayland ''
+    wrapProgramArgs+=(--prefix PATH : ${lib.makeBinPath [ xwayland ]})
+  '' + ''
+    wrapProgram $out/bin/cosmic-comp ''${wrapProgramArgs[@]}
```

I think this needs to be double quoted [to avoid having the shell reinterpret the arguments](https://www.shellcheck.net/wiki/SC2068)

```suggestion
    wrapProgram $out/bin/cosmic-comp "''${wrapProgramArgs[@]}"
```